junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/FileSource.java

```diff
@@ -55,26 +56,23 @@ public static FileSource from(File file) {
 	 * @param filePosition the position in the source file; may be {@code null}
 	 */
 	public static FileSource from(File file, @Nullable FilePosition filePosition) {
-		return new FileSource(file, filePosition);
+		Preconditions.notNull(file, "file must not be null");
+		try {
+			File canonicalFile = file.getCanonicalFile();
+			return new FileSource(canonicalFile, filePosition);
+		}
+		catch (IOException ex) {
+			throw new JUnitException("Failed to retrieve canonical path for file: " + file, ex);
+		}
 	}
 
 	private final File file;
 
 	@Nullable
 	private final FilePosition filePosition;
```

Comment:
It should probably be `private final @Nullable FilePosition filePosition` since the nullability belongs to a type rather than to a field

Reply:
Yep, you're right. I initially did it wrong and tried to replace them all using a pattern. Looks like I missed a few. I'll do another pass.

Reply:
Added to #4628